### PR TITLE
config: add http configuration

### PR DESCRIPTION
**Pull request by s-urbaniak**
Currently there is no possibility to add top-level HTTP configuration
to the operator.

This fixes it and applies the HTTP configuration to the oauth proxy
sidecar for alertmanager.

Fixes MON-255

### CODE REVIEW

File: pkg/manifests/manifests_test.go [hunk 1]
```diff
@@ -276,6 +278,131 @@ func TestUnconfiguredManifests(t *testing.T) {
...
+					if a.Spec.Containers[c].Env[e].Name != name {
+						continue
+					}
+					if a.Spec.Containers[c].Env[e].Value == value {
+						return nil
+					}
+				}
+			}
+
+			return fmt.Errorf("containers expected to have env name %v=%v, but they don't", name, value)
```
Comment (squat) [inline pkg/manifests/manifests_test.go]: should `env name` here actually be `env var`?

Comment (s-urbaniak) [reply pkg/manifests/manifests_test.go]: right :+1: it should actually be `name env=var`, I suggest simply to reduce to `containers expected to have env %v=%v, but they don't`

Comment (s-urbaniak) [reply pkg/manifests/manifests_test.go]: ah, sorry, your suggestion actually makes perfect sense.

File: pkg/manifests/manifests_test.go [hunk 2]
```diff
@@ -276,6 +278,131 @@ func TestUnconfiguredManifests(t *testing.T) {
...
+				}
+			}
+			return nil
+		}
+	}
+
+	hasContainers := func(expected bool) checkFunc {
+		return func(a *monv1.Alertmanager) error {
+			if got := len(a.Spec.Containers) > 0; got != expected {
+				return fmt.Errorf("expected spec to have containers %t, got %t", expected, got)
```
Comment (squat) [inline pkg/manifests/manifests_test.go]: small doubt: what would it mean for the error to say `expected spec to have containers false`? (or `true` for that matter)

Comment (s-urbaniak) [reply pkg/manifests/manifests_test.go]: yes, that would be the meaning ... hmm, do you have a better wording? :thinking:

Comment (squat) [reply pkg/manifests/manifests_test.go]: blah nothing short comes to mind. we could make it a bit more verbose and interpretable, like:
```go
var negative string
if !expected {
    negative = "not"
}
fmt.Errorf(strings.Join(" ", "expected spec", negative, "to have containers, got %d containers"), len(a.Spec.Containers))
```

But this is not necessary IMO

Comment (s-urbaniak) [reply pkg/manifests/manifests_test.go]: suggestion: as I am never asserting `hasContainers(false)` I can omit the parameter, always expecting `true`, which makes it more reasonable for now. Let's go with that.

```
	hasContainers := func() checkFunc {
		return func(a *monv1.Alertmanager) error {
			if len(a.Spec.Containers) == 0 {
				return errors.New("expected spec to have containers, but it doesn't have any")
			}
			return nil
		}
	}
```

**Pull request merged**